skfp/distances/russell.py

```diff
@@ -0,0 +1,202 @@
+from typing import Union
+
+import numpy as np
+from scipy.sparse import csr_array
+from sklearn.utils._param_validation import validate_params
+
+
+@validate_params(
+    {
+        "vec_a": [
+            "array-like",
+            csr_array,
+        ],
+        "vec_b": [
+            "array-like",
+            csr_array,
+        ],
+    },
+    prefer_skip_nested_validation=True,
+)
+def russell_binary_similarity(
+    vec_a: Union[
+        np.ndarray,
+        csr_array,
+    ],
+    vec_b: Union[
+        np.ndarray,
+        csr_array,
+    ],
+) -> float:
+    r"""
+    Russell similarity for vectors of binary values.
+
+    Computes the Russell similarity [1]_ [2]_ [3]_ for binary data between two
+    input arrays or sparse matrices, using the formula:
+
+    .. math::
+
+        sim(x, y) = \frac{a}{n}
+
+    where
+
+    - :math:`a` - common "on" bits.
+    - :math:`n` - length of passed vectors
+
+    The calculated similarity falls within the range :math:`[0, 1]`.
+    Passing all-zero vectors to this function results in a similarity of 0.
+    Passing empty vectors results in similarity of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    -------
+    similarity : float
+        Russell similarity between ``vec_a`` and ``vec_b``.
+
+    References
+    ----------
+    .. [1] `Russell P.F., Rao T.R.
+        "On habitat and association of species of anopheline larvae in south-eastern Madras. (1940)"
+        <https://www.cabidigitallibrary.org/doi/full/10.5555/19412900343>`_
+
+    .. [2] `Deza M.M., Deza E.
+        "Encyclopedia of Distances."
+        Springer, Berlin, Heidelberg, 2009.
+        <https://doi.org/10.1007/978-3-642-00234-2_1>`_
+
+    .. [3] `RDKit documentation
+        <https://www.rdkit.org/docs/source/rdkit.DataStructs.cDataStructs.html>`_
+
+    Examples
+    --------
+    >>> from skfp.distances import russell_binary_similarity
+    >>> import numpy as np
+    >>> vec_a = np.array([1, 1, 1])
+    >>> vec_b = np.array([1, 1, 1])
```

Comment:
It would be useful to include some less trivial example here, e.g. [1,1,1,1] and [1,1,0,0]